pkg/volume/projected/projected.go

```diff
@@ -350,7 +352,12 @@ func (c *projectedVolumeUnmounter) TearDownAt(dir string) error {
 	if err != nil {
 		return err
 	}
-	return wrapped.TearDownAt(dir)
+	if err = wrapped.TearDownAt(dir); err != nil {
+		return err
+	}
+
+	c.plugin.deleteServiceAccountToken(c.podUID)
```

Comment:
Only call this if there is a `ServiceAccountToken` source.

Reply:
@tallclair  Currently we can't not get the volume's detail directly in projectedVolumeUnmounter. As far as I can see, we need record projected volume's information in projectedPlugin each time NewMounter() was invoked, so projectedVolumeUnmounter could use that information to do check.
I think the negative impact here is limited if we just delete tokens without serviceaccount source bounded check.
WDYT

Reply:
```suggestion
func hasServiceAccountToken(sources []v1.VolumeProjection) bool {
  for _, src := range sources {
    if src.ServiceAccountToken != nil {
      return true
    }
  }
  return false
}
// ...
if hasSAToken(c.sources) {
  c.plugin.deleteServiceAccountToken(c.podUID)
}
```
Am I missing something?

Reply:
@tallclair When projectedVolumeMounter was constructed, the sources wasn't filled. And I have confirmed that the code you gave is not reasonable because c.sources is always empty.
https://github.com/kubernetes/kubernetes/blob/cbb422c214dd70735258b8e0af3b721613c2f188/pkg/volume/projected/projected.go#L123

Reply:
I see. Seems like the projectedVolume type should not be embedded then, but that's unrelated to this PR.
Thanks for clarifying.